src/stdlib_linalg_constants.fypp

```diff
@@ -0,0 +1,23 @@
+#:include "common.fypp"
+module stdlib_linalg_constants
+     use stdlib_kinds, only: sp, dp, qp, int32, int64, lk
+     use, intrinsic :: ieee_arithmetic, only: ieee_is_nan 
+#if defined(_OPENMP)
+     use omp_lib
```

Comment:
```suggestion
     !$ use omp_lib
```
Using "!$" will make this line compiled conditionnaly on the support of OpenMP by the compiler (and the CPP dir. is therefore not needed)

Reply:
Thank you @jvdp1, interesting! These `#if defined(_OPENMP)` statements were in the original LAPACK and I left them unchanged. Is it standard Fortran that they should be always read if they start with `!$`? If so, I could update them all

Reply:
This is intriguing, what I had understood is that conditional directives require specific signatures such as `!$omp` or `!$dir`, etc. But this `!$ something` is still just a comment... no?

Reply:
`!$ something` is a conditional compilation sentinel and part of the OpenMP API (see section 3.3.2 of [the OpenMP specs](https://www.openmp.org/spec-html/5.2/openmpsu26.html#x49-480003.3.2)).

@perazz It is indeed part of OpenMP API specs (at least, I think, since OpenMP 2.5). I use it since many years with both Intel and gfortran compilers. I guess that it is also supported by other compilers. However, note the I am not against the cpp directive. I just find using !$ cleaner ;)

Reply:
Wow, didn't know that!! just tested a small code using this with nvfortran 24.1 and works as expected!